tests/test_packages/test_connections/test_p2p_libp2p/base.py

```diff
@@ -0,0 +1,113 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2022 Valory AG
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""Base classes for p2p_libp2p tests"""
+
+import atexit
+import logging
+import os
+from pathlib import Path
+from typing import List
+
+from aea.mail.base import Envelope
+from aea.multiplexer import Multiplexer
+
+from packages.fetchai.protocols.default.message import DefaultMessage
+
+from tests.conftest import TEMP_LIBP2P_TEST_DIR, remove_test_directory
+
+
+class BaseP2PLibp2pTest:
+    """Base class for ACN p2p libp2p tests"""
+
+    cwd: str
+    t: str
+    tmp_dir: str
+    tmp_client_dir: str
+    log_files: List[str] = []
+    multiplexers: List[Multiplexer] = []
+    capture_log = True
+
+    @classmethod
+    def setup_class(cls):
+        """Set the test up"""
+
+        atexit.register(cls.teardown_class)
+        cls.cwd, cls.t = os.getcwd(), TEMP_LIBP2P_TEST_DIR
+        if Path(cls.t).exists():
+            cls.remove_temp_test_dir()
+        cls.tmp_dir = os.path.join(cls.t, "temp_dir")
+        cls.tmp_client_dir = os.path.join(cls.t, "temp_client_dir")
+        Path(cls.tmp_dir).mkdir(parents=True, exist_ok=True)
+        os.chdir(cls.t)
```

Comment:
first things first: register teardown. Use a predefined TEMP_LIBP2P_TEST_DIR and ensure it is removed before setting up the test.